pkg/controller/extension/actuator.go

```diff
@@ -172,6 +172,18 @@ func (a *actuator) Migrate(_ context.Context, _ logr.Logger, _ *extensionsv1alph
 	return nil
 }
 
+// ForceDelete the Extension resource.
```

Comment:
I would vote to add 1-2 sentences about the force deletion specifics in the docs string such as:

```suggestion
// ForceDelete the Extension resource.
//
// We don't need to wait for the ManagedResource deletion because ManagedResources are finalized by gardenlet
// in later step in the Shoot force deletion flow.
```